cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp

```diff
@@ -0,0 +1,161 @@
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/rbf_kernel/compute.hpp"
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/dataframe.hpp"
+#include "oneapi/dal/test/engine/math.hpp"
+
+namespace oneapi::dal::rbf_kernel::test {
+
+namespace te = dal::test::engine;
+namespace la = te::linalg;
+
+template <typename TestType>
+class rbf_kernel_batch_test : public te::algo_fixture {
+public:
+    using Float = std::tuple_element_t<0, TestType>;
+    using Method = std::tuple_element_t<1, TestType>;
+
+    auto get_descriptor(double sigma) const {
+        return rbf_kernel::descriptor<float, Method>{}.set_sigma(sigma);
+    }
+
+    te::table_id get_homogen_table_id() const {
+        return te::table_id::homogen<Float>();
+    }
+
+    void general_checks(const te::dataframe& x_data,
+                        const te::dataframe& y_data,
+                        double sigma,
+                        const te::table_id& x_data_table_id,
+                        const te::table_id& y_data_table_id) {
+        CAPTURE(sigma);
+        const table x = x_data.get_table(x_data_table_id);
+        const table y = y_data.get_table(y_data_table_id);
+
+        INFO("create descriptor")
+        const auto rbf_kernel_desc = get_descriptor(sigma);
+
+        INFO("run compute");
+        const auto compute_result = this->compute(rbf_kernel_desc, x, y);
+        check_compute_result(sigma, x, y, compute_result);
+    }
+
+    void check_compute_result(double sigma,
+                              const table& x_data,
+                              const table& y_data,
+                              const rbf_kernel::compute_result<>& result) {
+        const auto result_values = result.get_values();
+
+        SECTION("result values table shape is expected") {
+            REQUIRE(result_values.get_row_count() == x_data.get_row_count());
+            REQUIRE(result_values.get_column_count() == y_data.get_row_count());
+        }
+
+        SECTION("there is no NaN in result values table") {
+            REQUIRE(te::has_no_nans(result_values));
+        }
+
+        SECTION("result values are expected") {
+            check_result_values(sigma, x_data, y_data, result_values);
+        }
+    }
+
+    void check_result_values(double sigma,
+                             const table& x_data,
+                             const table& y_data,
+                             const table& result_values) {
+        auto reference = compute_reference(sigma, x_data, y_data);
+        const double tol = te::get_tolerance<Float>(1e-2, 1e-2);
+        const double diff = la::l_inf_norm(reference, la::matrix<double>::wrap(result_values));
+        CHECK(diff < tol);
+    }
+
+    la::matrix<double> compute_reference(double sigma, const table& x_data, const table& y_data) {
```

Comment:
I would suggest using a exactly formula without gemm: exp(-gamma * ||x - y||)

See reference in new_ts. 

Gemm appears as a result of not exactly the formula: exp(-gamma * (||x||  - **2(x,y)** + ||y||))